dlib/cuda/cpu_dlib.h

```diff
@@ -442,12 +442,14 @@ namespace dlib
         inline void resize_bilinear (
             tensor& dest,
             const tensor& src
-        ) { resize_bilinear(dest, dest.nc(), dest.nr()*dest.nc(), src, src.nc(), src.nr()*src.nc()); }
+        ) { resize_bilinear(dest, static_cast<long>(dest.nc()), static_cast<long>(dest.nr()*dest.nc()),
```

Comment:
This is kinda funky.  What's the warning? Visual studio is complaining about loss of precision going from `long long` to `long` presumably?  That's a silly warning honestly.  A ton of code performs this kind of implicit casting.

Anyway, it would make more sense to change `resize_bilinear` to take `long long` though than add these casts.  Can you do that instead?

Reply:
Hmm. For some reason, when I now reverted these changes in my local build, Visual Studio no longer complains about any of these issues. So it doesn't look like I need any of these changes after all. This is strange, because I do not admit having changed any compiler warning level settings in between. (The complaining started, when I upgraded my project from VS 2017 to VS 2019.)

So I'll at least revert these overly verbose `static_cast`s. But I guess there are other small changes here that could be considered minor improvements. And yes, maybe I'll also make `resize_bilinear` take `long long`, while at it?

Reply:
Yeah it's sensible to make `resize_bilinear` use `long long` to match what `tensor` uses.  If you want to adjust that that's cool :)